internal/webhook/module.go

```diff
@@ -141,6 +149,19 @@ func validateModuleLoaderContainerSpec(container kmmv1beta1.ModuleLoaderContaine
 			(km.InTreeModuleToRemove != "" && container.InTreeModulesToRemove != nil) { //nolint:staticcheck
 			return fmt.Errorf("only one type if field (InTreeModuleToRemove or InTreeModulesToRemove) can be defined in KenrelMapping or Container")
 		}
+
+		var imgToCheck string
+		imgToCheck = container.ContainerImage
+		if km.ContainerImage != "" {
+			imgToCheck = km.ContainerImage
+		}
+		if !strings.ContainsRune(imgToCheck, '@') && !strings.ContainsRune(imgToCheck, ':') {
```

Comment:
why can't we just return ImageHashOrTagNotSet from here? Why do we have to wait for the loop to finish?

Reply:
Why do we need a warning in the webhook? Container image is either valid or not.

Reply:
I think we should have a new mutating webhook that sets the `latest` tag instead of returning warnings.